internal/mock/overlay.go

```diff
@@ -1,11 +1,11 @@
 // Copyright (C) 2018 Storj Labs, Inc.
 // See LICENSE for copying information.
 
-package overlay
+package mock
```

Comment:
Rather than moving to `internal/mock` move it to `pkg/overlay/mockoverlay`.

Reply:
why? I think it makes more sense to keep mocks all in one place.

Reply:
The problem with collecting all mocks (or any specific category of things) into one place is that we might end up with 50 different mocks in a single folder, eventually. The locality of changes also suffers as well. Rather than changing one package and a subpackage, you end up changing packages with a larger distance (a similar problem with regards to documentation as an example http://wiki.c2.com/?LocalityOfReferenceDocumentation)

Reply:
done